### PR TITLE
Emit more relevant error message when failing to add source folder

### LINKED ISSUE #1020: Adding source files to workspace fails with "couldn't obtain file explorer uri for \<dbname\>"
**Describe the bug**
When trying to add sources from DB to Visual Studio workspace, the output log indicates a 
"Couldn't obtain file explorer uri for \<dbname\>" 
message, but the sources are not added to the file explorer view.
Queries work like a charm and are linked to sources that will open as expected.

The issue occurs on a bundled DB zip archive as well as on an unzipped archive that was added to VS as folder.
The db zip includes both, a src folder and an src.zip.
The dbbundle has been created using 
codeql database create --language=cpp --command=make \<dbname\>
codeql database bundle --output=\<dbname\>.zip --include-uncompressed-source \<dbname\>
on a Linux host using codeQL CLI 2.7.0
VS Code is running on a Windows host in version 1.62.3 and CodeQL Extension 1.5.7 using CodeQL CLI 2.7.2

**Version**
VS Code is running on a Windows host in version 1.62.3 and CodeQL Extension 1.5.7 using CodeQL CLI 2.7.2
DB was created on a Linux host using codeQL CLI 2.7.0

**To reproduce**
create DB as above on linux host
copy bundled DB to analyzing windows host 
start vscode-codeql-starter workspace
add database from archive (or from folder after unzipping it)
rightclick on database in codeql databases section
pick "add database source to workspace"

**Expected behavior**
Source files are added to workspace and are visible in file explorer

### PR DESCRIPTION
**Pull request by aeisenberg**
Fixes #1020

Replace this with a description of the changes your pull request makes.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/databases.ts [hunk 1]
```diff
@@ -459,13 +464,26 @@ export class DatabaseItemImpl implements DatabaseItem {
...
   }
 
+  public verifyZippedSources(): string | undefined {
+    const sourceArchive = this.sourceArchive;
+    if (sourceArchive === undefined) {
+      return `${this.name} has no source archive.`;
+    }
+
+    if (!sourceArchive.fsPath.endsWith('.zip')) {
+      return `${this.name} has a source archive that is unzipped.`;
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases.ts:482]:
```suggestion
      return `${this.name} has a source folder that is unzipped.`;
```
Minor (is it still an archive if it's unzipped?)

File: extensions/ql-vscode/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## [UNRELEASED]
 
+- Emit more explicit error message when a user tries to add unzipped sources to the workspace. [1021](https://github.com/github/vscode-codeql/pull/1021)
```
Comment (shati-patel) [inline extensions/ql-vscode/CHANGELOG.md:5]: Maybe something like this, to clarify what these "sources" are? 
```suggestion
- Emit a more explicit error message when a user tries to add a database with an unzipped source folder to the workspace. [1021](https://github.com/github/vscode-codeql/pull/1021)
```

**Review by shati-patel — APPROVED** [review]
Thanks, looks good! 🎉 Just a few suggestions on the wording 📝

**Comment by aeisenberg** [comment]
@shati-patel Thanks for the clarifications!

**Pull request merged**